crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4011,51 +3993,49 @@ impl<'db> IterationErrorKind<'db> {
+                        because it may not have an `__iter__` method \
+                        and its `__getitem__` attribute has type `{dunder_getitem_type}`, \
+                        which is not callable",
+                    iterable_type = iterable_type.display(db),
+                    dunder_getitem_type = bindings.callable_type.display(db),
+                )),
+                CallDunderError::CallError(CallErrorKind::PossiblyNotCallable, bindings) if bindings.is_single() => report_not_iterable(format_args!(
+                    "Object of type `{iterable_type}` may not be iterable \
+                        because it may not have an `__iter__` method \
+                        and its `__getitem__` attribute may not be callable",
+                    iterable_type = iterable_type.display(db),
+                )),
+                CallDunderError::CallError(CallErrorKind::PossiblyNotCallable, bindings) => {
+                    report_not_iterable(format_args!(
                         "Object of type `{iterable_type}` may not be iterable \
                             because it may not have an `__iter__` method \
-                            and its `__getitem__` method (with type `{dunder_getitem_type}`)
-                            may have an incorrect signature for the old-style iteration protocol \
-                            (expected a signature at least as permissive as \
-                            `def __getitem__(self, key: int): ...`)",
+                            and its `__getitem__` attribute (with type `{dunder_getitem_type}`) \
+                            may not be callable",
                         iterable_type = iterable_type.display(db),
-                        dunder_getitem_type = called_type.display(db),
-                    )),
+                        dunder_getitem_type = bindings.callable_type.display(db),
+                    ));
                 }
+                CallDunderError::CallError(CallErrorKind::BindingError, bindings) if bindings.is_single() => report_not_iterable(format_args!(
+                    "Object of type `{iterable_type}` may not be iterable \
+                        because it may not have an `__iter__` method \
+                        and its `__getitem__` method has an incorrect signature \
+                        for the old-style iteration protocol \
+                        (expected a signature at least as permissive as \
+                        `def __getitem__(self, key: int): ...`)",
+                    iterable_type = iterable_type.display(db),
+                )),
+                CallDunderError::CallError(CallErrorKind::BindingError, bindings) => report_not_iterable(format_args!(
+                    "Object of type `{iterable_type}` may not be iterable \
+                        because it may not have an `__iter__` method \
+                        and its `__getitem__` method (with type `{dunder_getitem_type}`)
```

Comment:
oops -- the change to the snapshot makes me realise that there was a pre-existing bug here; this should be

```suggestion
                        and its `__getitem__` method (with type `{dunder_getitem_type}`) \
```

Reply:
Thanks!